components/service/glean/src/main/java/mozilla/components/service/glean/BooleanMetricType.kt

```diff
@@ -58,8 +62,16 @@ data class BooleanMetricType(
      *                 value in [sendInPings]
      * @return true if metric value exists, otherwise false
      */
+    @Suppress("MagicNumber")
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testHasValue(pingName: String = getStorageNames().first()): Boolean {
+        if (ioTask != null) {
+            var loopCnt = 50
+            while (ioTask!!.isActive && loopCnt > 0) {
+                --loopCnt
```

Comment:
Would it help to have a short "sleep" here?  I'm sure Java is different, but at least on Python, if you deadspin while waiting on another thread like this, you're much less likely to give the other thread a chance to finish, so `sleep`'ing can help.

Reply:
I considered that, but Thread.sleep() is usually frowned upon for busy loops.  I totally agree that relinquishing the processor would be a good idea here, maybe since this is only for tests it would be acceptable to have a 5 ms sleep or something?